PowerEditor/src/ScintillaComponent/AutoCompletion.cpp

```diff
@@ -458,7 +458,10 @@ void AutoCompletion::getWordArray(vector<generic_string> & wordArray, TCHAR *beg
 
 	size_t docLength = _pEditView->execute(SCI_GETLENGTH);
 
-	int flags = SCFIND_WORDSTART | SCFIND_MATCHCASE | SCFIND_REGEXP | SCFIND_POSIX;
+	int flags = SCFIND_WORDSTART | SCFIND_REGEXP | SCFIND_POSIX;
+
+	if (!_ignoreCase || (!_funcCompletionActive && _curLang == L_TEXT))

```

Comment:
Hmm... if current lang is L_TEXT, `_funcCompletionActive` should be false, since there's no function xml to be load.
Is `!_funcCompletionActive` condition necessary?

Reply:
If *normal.xml* exist, then `_funcCompletionActive` will be true so then the xml settings will apply. If `_funcCompletionActive ` is not used there, then the xml settings will be overruled there and text would be always searched as case sensitive. Having `_funcCompletionActive` there prevents *normal.xml* Environment options being ignored.

*normal.xml* is mentioned [here](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/12325#issue-1406561749) which made me aware of *normal.xml* .

Reply:
If the condition seems too confusing, then could change to only

```c++
int flags = SCFIND_WORDSTART | (_ignoreCase ? SCFIND_NONE : SCFIND_MATCHCASE) | SCFIND_REGEXP | SCFIND_POSIX;
```

though text would be case insensitive by default. I was treating text as special so that both `The` and `the` kind of word variations does not polute the list.

Reply:
@mpheath 
You're right about it. I forgot "normal.xml" case could happen.
So your original version is good. Nothing to be changed.